x-pack/packages/kbn-elastic-assistant/impl/assistant/prompt_editor/system_prompt/select_system_prompt/index.tsx

```diff
@@ -134,38 +97,9 @@ const SelectSystemPromptComponent: React.FC<Props> = ({
         setSelectedSettingsTab(SYSTEM_PROMPTS_TAB);
         return;
       }
-      // Note: if callback is provided, this component does not persist. Extract to separate component
-      if (onSystemPromptSelectionChange != null) {
-        onSystemPromptSelectionChange(selectedSystemPromptId);
```

Comment:
There are 2 places the `SelectSystemPrompt` component is implemented: `ConversationSettingsEditor` and `EmptyConvo` (via `SystemPrompt`)

Previously, the `ConversationSettingsEditor` passed this method and then handled the system prompt API calls and subsequent UI updates. For some silly reason, it also passed in the UI methods. API call 1 was made below to update the conversation immediately. Then if in the modal the user pressed "Cancel", API call 2 is made to reset the conversation. API call 1 is unnecessary.

`EmptyConvo` did not pass `onSystemPromptSelectionChange` and relied on `SelectSystemPrompt` to do the update. I moved the update function (`setSelectedSystemPrompt `) to `useCurrentConverstion` to keep the API calls and conversation updates out of the UI components. So now `EmptyConvo` passes `onSystemPromptSelectionChange` and the prompt change is handled upstream.